### PR TITLE
[LOG_FORWARDER] [REDIS] Support for queues, sets, publish and streams modes for log insertion

### PR DESCRIPTION
**Pull request by ELDiablO59152**
### Added
- [LOG_FORWARDER] [REDIS] Support for queues, sets, publish and streams modes for log insertion

### CODE REVIEW

File: vulture_os/applications/logfwd/models.py [hunk 1]
```diff
@@ -419,10 +426,40 @@ def to_template(self, **kwargs):
...
+    expire_key = models.PositiveIntegerField(
+        default=0,
+        blank=True,
+        help_text=_("Use SETX instead of SET in key mode with an expiration in seconds"),
+        verbose_name=_("Expiration of the key (s)"),
+    )
+    stream_outfield = models.TextField(
+        default="msg",
+        blank=True,
+        help_text=_("Set the name of the log index in stream mode"),
```
Comment (frikilax) [inline vulture_os/applications/logfwd/models.py:454]:
```suggestion
        help_text=_("Set the name of the index field to use when inserting log, in stream mode"),
```

File: vulture_os/applications/logfwd/models.py [hunk 2]
```diff
@@ -419,10 +426,40 @@ def to_template(self, **kwargs):
...
+        verbose_name=_("Index name of the log"),
+    )
+    stream_capacitylimit = models.PositiveIntegerField(
+        default=0,
+        blank=True,
+        help_text=_("Set a maximum size of a stream to prevent out of range index"),
```
Comment (frikilax) [inline vulture_os/applications/logfwd/models.py:460]:
```suggestion
        help_text=_("Set a stream capacity limit, if set to more than 0 (zero), oldest values in the stream will be evicted to stay under the max value"),
```

File: vulture_os/applications/logfwd/config/om_hiredis.tpl [hunk 3]
```diff
@@ -11,6 +11,18 @@
...
+        {%- endif %}
+        {%- if mode == "set" %}
+            Expiration="{{expire_key}}"
+        {%- endif %}
+        {%- if mode == "stream" %}
+            stream.outField="{{stream_outfield}}"
+        {%- endif %}
+        {%- if mode == "stream" %}
+            stream.capacityLimit="{{stream_capacitylimit}}"
         {%- endif %}
```
Comment (frikilax) [inline vulture_os/applications/logfwd/config/om_hiredis.tpl:26]:
```suggestion
        {%- if mode == "stream" %}
            stream.outField="{{stream_outfield}}"
            stream.capacityLimit="{{stream_capacitylimit}}"
        {%- endif %}
```

File: vulture_os/applications/logfwd/models.py [hunk 4]
```diff
@@ -419,10 +426,40 @@ def to_template(self, **kwargs):
...
+    use_rpush = models.BooleanField(
+        default=False,
+        blank=True,
+        help_text=_("Use RPUSH instead of LPUSH in list mode"),
+        verbose_name=_("Use RPUSH"),
+    )
+    expire_key = models.PositiveIntegerField(
+        default=0,
+        blank=True,
+        help_text=_("Use SETX instead of SET in key mode with an expiration in seconds"),
```
[the remark this answers is not in the document]
Comment (ELDiablO59152) [reply vulture_os/applications/logfwd/models.py]: Done

File: vulture_os/applications/logfwd/models.py (hunk 1 above)
Comment (ELDiablO59152) [reply vulture_os/applications/logfwd/models.py]: Done

File: vulture_os/applications/logfwd/config/om_hiredis.tpl (hunk 3 above)
Comment (ELDiablO59152) [reply vulture_os/applications/logfwd/config/om_hiredis.tpl]: Done

**Review by frikilax — APPROVED** [review]

**Pull request merged**